### ISSUE TITLE
Error when using `file-paths` together with subfolder

### ISSUE THREAD

**Issue by marianfoo** [SAP/ui5-linter#259]
## Expected Behavior

Using `file-paths` and wildcards would check the folder plus all subfolders

## Current Behavior

ui5-linter errors out because logic thinks subfolder is a file.
 Using this cli command: `ui5lint --file-paths webapp/controller/* webapp/model/* webapp/view/*`

And this folder structure:
![image](https://github.com/user-attachments/assets/b46fe790-d770-4d98-a2a5-af310aeb2f26)

Works without `file-paths`.  
I´d rather use something to exclude folders or filepaths but currently i have to include all the folders i would like to test.
It would be much easier to just exclude the `test` folder.  
In a CI pipeline i need a either it´s all good or not. Doing it manually is no problem, i can just ignore it.

## Steps to Reproduce the Issue

```bash
git clone https://github.com/marianfoo/ui5linterbug
cd ui5linterbug
npm i
npm run lint
```

## Context

- UI5 linter version: **`0.3.2`**
- Node.js Version: **`20.14.0`**
- npm Version: `10.3.0`
- OS/Platform: `Mac`

## Log Output / Stack Trace

```
npm run lint -- --verbose

> com.myorg.myapp@1.0.0 lint
> ui5lint --file-paths webapp/controller/* webapp/model/* webapp/view/* --verbose

verb cli:middlewares:base using ui5lint version 0.3.2 (from /Users/marianzeis/DEV/ui5linterbug/node_modules/@ui5/linter/bin/ui5lint.js)
verb cli:middlewares:base using node version v20.14.0
verb generateProjectGraph Creating project graph using object...
verb graph:Module Could not find any build manifest in module ui5-linter-target
verb graph:Module Configuration for module ui5-linter-target is provided in YAML file at /Users/marianzeis/DEV/ui5linterbug/ui5.yaml
verb specifications:types:Application Path mapping for application project com.myorg.myapp:
verb specifications:types:Application   Physical root path: /Users/marianzeis/DEV/ui5linterbug
verb specifications:types:Application   Mapped to: webapp
verb specifications:types:Application Namespace of project com.myorg.myapp is com/myorg/myapp (from manifest.json)
verb graph:Module Module ui5-linter-target contains project com.myorg.myapp
verb graph:projectGraphBuilder Root project com.myorg.myapp qualified as application project for project graph
verb graph:ProjectGraph Adding project: com.myorg.myapp
verb graph:ProjectGraph Skipping resolution of optional dependencies since none have been declared
perf Trace Project Graph creation took 112.17 ms
perf Trace Linting manifest.json files took 0.15 ms
perf Trace XML Transpiler initialization took 0.95 ms
perf Trace Transpile XML took 16.31 ms - /resources/com/myorg/myapp/view/Main.view.xml
perf Trace Transpile XML took 17.15 ms - /resources/com/myorg/myapp/view/App.view.xml

⚠️  Process Failed With Error

Error Message:
File not found: /resources/com/myorg/myapp/controller/subfolder.ts

Stack Trace:
Error: File not found: /resources/com/myorg/myapp/controller/subfolder.ts
    at Object.getSourceFile (file:///Users/marianzeis/DEV/ui5linterbug/node_modules/@ui5/linter/lib/linter/ui5Types/host.js:188:23)
    at findSourceFileWorker (/Users/marianzeis/DEV/ui5linterbug/node_modules/@ui5/linter/node_modules/typescript/lib/typescript.js:125889:23)
    at findSourceFile (/Users/marianzeis/DEV/ui5linterbug/node_modules/@ui5/linter/node_modules/typescript/lib/typescript.js:125805:20)
    at /Users/marianzeis/DEV/ui5linterbug/node_modules/@ui5/linter/node_modules/typescript/lib/typescript.js:125754:22
    at /Users/marianzeis/DEV/ui5linterbug/node_modules/@ui5/linter/node_modules/typescript/lib/typescript.js:125746:91
    at forEach (/Users/marianzeis/DEV/ui5linterbug/node_modules/@ui5/linter/node_modules/typescript/lib/typescript.js:2388:22)
    at getSourceFileFromReferenceWorker (/Users/marianzeis/DEV/ui5linterbug/node_modules/@ui5/linter/node_modules/typescript/lib/typescript.js:125746:44)
    at processSourceFile (/Users/marianzeis/DEV/ui5linterbug/node_modules/@ui5/linter/node_modules/typescript/lib/typescript.js:125752:5)
    at processRootFile (/Users/marianzeis/DEV/ui5linterbug/node_modules/@ui5/linter/node_modules/typescript/lib/typescript.js:125544:5)
    at /Users/marianzeis/DEV/ui5linterbug/node_modules/@ui5/linter/node_modules/typescript/lib/typescript.js:124269:41
```

**Comment by flovogt** [comment]
@marianfoo thanks for reporting this issue. We will fix this bug in backlog item CPOUI5FOUNDATION-881.

**Comment by flovogt** [comment]
Fixed with https://github.com/SAP/ui5-linter/pull/312
